src/v/config/configuration.cc

```diff
@@ -2373,6 +2373,14 @@ configuration::configuration()
       false)
   , cloud_storage_disable_metadata_consistency_checks(
       *this, "cloud_storage_disable_metadata_consistency_checks")
+  , cloud_storage_disable_archival_stm_rw_fence(
```

Comment:
I don't see any tests using this so why are we introducing it? At last, this should be a `development_feature_property` recently introduced by @dotnwat